File: C++/20. Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> ans;
        map<char, char> maple;
        maple[']'] = '[';
        maple['}'] = '{';
        maple[')'] = '(';
        for(auto c: s)
        {
            if(c=='{' || c=='(' || c=='[')
                ans.push(c);
            else if(ans.empty() || ans.top() != maple[c])
                return false;
            else
                ans.pop();
        }
        return ans.empty();
    }
};

File: Python/20. Valid Parentheses.py
class Solution:
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        ans = list()
        maple = {'}': '{', ')': '(', ']': '['}
        for c in s:
            if c in ['{', '[', '(']:
                ans.append(c)
            elif len(ans) == 0 or ans[-1] != maple[c]:
                return False
            else:
                ans.pop()
        return ans == []
                